src/repositories/PostRepository.ts

```diff
@@ -51,35 +55,61 @@ export class PostRepository extends AbstractRepository<PostModel> {
     post.categories = categories;
     post.price = price;
     post.images = images;
+    post.archive = false;
     post.user = user;
     return await this.repository.save(post);
   }
 
   public async deletePost(post: PostModel): Promise<PostModel> {
-    return this.repository.remove(post);
+    return await this.repository.remove(post);
   }
 
   public async searchPostsByTitle(keywords: string): Promise<PostModel[]> {
     return await this.repository
       .createQueryBuilder("post")
       .leftJoinAndSelect("post.user", "user")
-      .where("LOWER(post.title) like LOWER(:keywords)", {keywords: `%${keywords}%`})
+      .where("LOWER(post.title) like LOWER(:keywords)", { keywords: `%${keywords}%` })
+      .andWhere("post.archive = false")
       .getMany();
   }
 
   public async searchPostsByDescription(keywords: string): Promise<PostModel[]> {
     return await this.repository
       .createQueryBuilder("post")
       .leftJoinAndSelect("post.user", "user")
-      .where("LOWER(post.description) like LOWER(:keywords)", {keywords: `%${keywords}%`})
+      .where("LOWER(post.description) like LOWER(:keywords)", { keywords: `%${keywords}%` })
+      .andWhere("post.archive = false")
       .getMany();
   }
 
   public async filterPosts(category: string): Promise<PostModel[]> {
     return await this.repository
       .createQueryBuilder("post")
       .leftJoinAndSelect("post.user", "user")
-      .where(":category = ANY (post.categories)", {category: category})
+      .where(":category = ANY (post.categories)", { category: category })
+      .andWhere("post.archive = false")
       .getMany();
   }
+
+  public async getArchivedPosts(): Promise<PostModel[]> {
```

Comment:
a little confused with the occurrence of archived posts everywhere, is this what 'Add post deletion logic for admins' is? if so, can you make the distinction clearer in the PR description too

Reply:
they're unrelated. Im also confused why archived posts is everywhere when it was already merged in with my last pr